primo/demo/PRIMO - Example_1.ipynb

```diff
@@ -5242,86 +5667,748 @@
+       "      <th></th>\n",
+       "      <td>87</td>\n",
+       "      <td>5</td>\n",
+       "      <td>64.67</td>\n",
+       "      <td>50.06</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th></th>\n",
+       "      <td>100</td>\n",
+       "      <td>5</td>\n",
+       "      <td>69.35</td>\n",
+       "      <td>49.27</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th></th>\n",
+       "      <td>72</td>\n",
+       "      <td>7</td>\n",
+       "      <td>64.83</td>\n",
+       "      <td>48.62</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th></th>\n",
+       "      <td>58</td>\n",
+       "      <td>8</td>\n",
+       "      <td>65.85</td>\n",
+       "      <td>44.21</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th></th>\n",
+       "      <td>52</td>\n",
+       "      <td>5</td>\n",
+       "      <td>64.17</td>\n",
+       "      <td>43.99</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th></th>\n",
+       "      <td>55</td>\n",
+       "      <td>5</td>\n",
+       "      <td>62.20</td>\n",
+       "      <td>40.89</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       " Project  Number of Wells  Impact Score [0-100]  Efficiency Score [0-100]\n",
+       "      75                5                 63.23                     57.24\n",
+       "      62               11                 65.45                     51.32\n",
+       "      87                5                 64.67                     50.06\n",
+       "     100                5                 69.35                     49.27\n",
+       "      72                7                 64.83                     48.62\n",
+       "      58                8                 65.85                     44.21\n",
+       "      52                5                 64.17                     43.99\n",
+       "      55                5                 62.20                     40.89"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Total budget recommended for plugging oil wells is: 4,284,000\n"
+     ]
+    }
+   ],
+   "source": [
+    "violation_oil = Recalculate(selected_oil, well_add_list_oil, well_remove_list_oil,oil_gdf,opt_inputs_oil)\n",
+    "\n",
+    "violate_cost_oil=violation_oil.assess_budget()\n",
+    "violate_operator_oil=violation_oil.assess_owner_well_count()\n",
+    "violate_distance_oil=violation_oil.assess_distances()\n",
+    "\n",
+    "if violate_cost_oil > 0:\n",
+    "    display(Markdown(f\"\"\" After the modification, the total budget is over the limit by ${int(violate_cost_oil)}.\n",
+    "                     Please consider to modify wells you have selected use the widget above or re-run the optimization problem.\"\"\"))\n",
```

Comment:
"Please consider modifying the wells you have selected by using either the widget above, or by re-running the optimization problem"

Reply:
Fixed!